From 088975a020912b19494f4776905ea01fc51082d3 Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date: Sun, 6 Mar 2011 20:26:09 +0100
Subject: [PATCH 01/14] - honor PKG_CONFIG_DEBUG_SPEW, which is like our --debug option

---
 pkg-config   |    2 ++
 pkg-config.1 |    6 ++++++
 2 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/pkg-config b/pkg-config
index cb4f6a4..b8943df 100644
--- a/pkg-config
+++ b/pkg-config
@@ -55,6 +55,8 @@ my $D = 0; # debug flag
     }
 }
 
+$D = 1 if defined($ENV{PKG_CONFIG_DEBUG_SPEW});
+
 if ($logfile) {
 	open my $L, ">>" . $logfile;
 	print $L '[' . join('] [', $0, @ARGV) . "]\n";
diff --git a/pkg-config.1 b/pkg-config.1
index 0c329f6..7cc7483 100644
--- a/pkg-config.1
+++ b/pkg-config.1
@@ -128,6 +128,12 @@ This can be used to specify a colon-separated list of paths to search for
 package files.
 If given, this list of paths is prepended to the standard search path.
 .El
+.Bl -tag -width PKG_CONFIG_DEBUG_SPEW
+.It Ev PKG_CONFIG_DEBUG_SPEW
+If set, this will cause
+.Nm
+to print various debugging information.
+.El
 .Sh EXIT STATUS
 .Ex -std pkg-config
 .Sh SEE ALSO
-- 
1.7.3.5

