From 2ee49a938be52e0c0777f334d13cd57c2f9c8660 Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date: Tue, 8 Mar 2011 11:05:41 +0100
Subject: [PATCH 10/14] - add stubs for new functions which will be part of 0.24

---
 pkg-config   |    3 +++
 pkg-config.1 |    3 +++
 2 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/pkg-config b/pkg-config
index a12464d..be64b00 100644
--- a/pkg-config
+++ b/pkg-config
@@ -105,6 +105,9 @@ GetOptions(	'debug' => \$D,
 		'modversion' => \$mode{modversion},
 		'variable=s' => \$mode{variable},
 		'define-variable=s' => $variables,
+		'print-provides' => sub { return 0; },
+		'print-requires' => sub { return 0; },
+		'print-requires-private' => sub { return 0; },
 	);
 
 # Initial value of printerr depends on the options...
diff --git a/pkg-config.1 b/pkg-config.1
index 8bd5311..41db8e5 100644
--- a/pkg-config.1
+++ b/pkg-config.1
@@ -106,6 +106,9 @@ returns its own version.
 .It Cm --print-errors
 Print error messages and exit with an error when errors occur.
 This is the default operation.
+.It Cm --print-provides
+.It Cm --print-requires
+.It Cm --print-requires-private
 .It Cm --silence-errors
 Do not print error messages, just exit with an error.
 .It Cm --static
-- 
1.7.3.5

