From c36fd2fe9becbde0ed3d5b87704c524c0d0db410 Mon Sep 17 00:00:00 2001
From: Jasper Lievisse Adriaanse <jasper@humppa.nl>
Date: Tue, 8 Mar 2011 11:35:05 +0100
Subject: [PATCH 13/14] - minor comment nit

---
 pkg-config |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/pkg-config b/pkg-config
index b0a827f..8477655 100644
--- a/pkg-config
+++ b/pkg-config
@@ -373,7 +373,8 @@ sub do_variable
 	$rc = 1;
 }
 
-#if the modversion option is set, pull out the compiler flags
+#if the modversion or print-provides options are set,
+#pull out the compiler flags
 sub do_modversion
 {
 	my ($p) = @_;
-- 
1.7.3.5

